### PR TITLE
zdsp: Add conversion and printing utilities

### PR DESCRIPTION
**Pull request by yperess**
- Add conversion utilities for q format to floats and back
- Add printing utilities for printf style formatting using PRIq() for the string format and PRIq_arg() to generate the args matching it.

### CODE REVIEW

**Comment by yperess** [comment]
@huhebo these should help you shim the existing Intel sensor logic to the zDSP library.

File: include/zephyr/dsp/conversions.h [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+}
+
+/**
+ * @brief Convert Q7 format to float
+ *
+ * @param[in]  src The Q7 input vector
+ * @param[out] dst The float output vector
+ * @param[in]  count Number of elements in each vector
+ */
+static inline void zdsp_q7_to_float(const q7_t *src, float *dst, uint32_t count)
```
Comment (teburd) [inline include/zephyr/dsp/conversions.h:57]: Shouldn't these take a shift value to get the full range?

File: include/zephyr/dsp/conversions.h [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+}
+
+/**
+ * @brief Convert floats to Q15 format
+ *
+ * @param[in]  src The float input vector
+ * @param[out] dst The Q15 output vector
+ * @param[in]  count Number of elements in each vector
+ */
+static inline void zdsp_float_to_q15(const float *src, q15_t *dst, uint32_t count)
```
Comment (teburd) [inline include/zephyr/dsp/conversions.h:85]: Vice versa, shouldn't these return a shift value and/or possibly error if the float is larger than the q value can hold?

**Review by teburd — COMMENTED** [review]
Looks really nice, and the tests do seem to cover allowable ranges of floats.

I guess my questions are mostly going to be what happens with floats larger than -1:1, and vice versa q values representing larger ranges with known shifts.

File: include/zephyr/dsp/conversions.h (hunk 1 above)
Comment (huhebo) [reply include/zephyr/dsp/conversions.h:57]: Agree!

File: include/zephyr/dsp/conversions.h (hunk 2 above)
Comment (huhebo) [reply include/zephyr/dsp/conversions.h:85]: Agree!

Comment (yperess) [reply include/zephyr/dsp/conversions.h:85]: I've made the change this morning but didn't finish writing the extra tests. I was originally thinking that the caller will manage this, but I'm ok with it being part of the API.

**Review by teburd — APPROVED** [review]

**Review by huhebo — APPROVED** [review]

**Comment by yperess** [comment]
@stephanosio I'm sure you're swamped this week, but any chance you could take a look at this before the window closes?

**Review by stephanosio — CHANGES_REQUESTED** [review]
> CANCEL that, this isn't a good PR. I found https://github.com/zephyrproject-rtos/cmsis-dsp/blob/zephyr/Include/dsp/support_functions.h which is really what the functions should be mirroring.

Yes, please mirror the conversion functions from the CMSIS-DSP "support functions" instead of introducing a new API.

**Pull request closed without merging**